internal/provider/resource_vdb_test.go

```diff
@@ -58,6 +78,79 @@ func testAccCheckDctVDBConfigBasic() string {
 	`, datasource_id)
 }
 
+func testAccCheckDctVDBBookmarkConfigBasic() string {
+	var bookmark_id, vdb_id string
+
+	// init client
+	cfg := dctapi.NewConfiguration()
+	cfg.Host = os.Getenv("DCT_HOST")
+	cfg.Scheme = "https"
+	cfg.HTTPClient = &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}}
+	cfg.AddDefaultHeader("Authorization", "apk "+os.Getenv("DCT_KEY"))
+	client := dctapi.NewAPIClient(cfg)
+
+	// create vdb
+	provisionVDBBySnapshotParameters := dctapi.NewProvisionVDBBySnapshotParameters()
+	provisionVDBBySnapshotParameters.SetAutoSelectRepository(true)
+	provisionVDBBySnapshotParameters.SetSourceDataId(os.Getenv("DATASOURCE_ID"))
+
+	vdb_req := client.VDBsApi.ProvisionVdbBySnapshot(context.Background())
+
+	vdb_res, vdb_http_res, vdb_err := vdb_req.ProvisionVDBBySnapshotParameters(*provisionVDBBySnapshotParameters).Execute()
+	if diags := apiErrorResponseHelper(vdb_res, vdb_http_res, vdb_err); diags != nil {
+		println("An error occured during vdb creation: " + vdb_err.Error())
+		return "" // return empty config to indicate config error
+	}
+	vdb_id = *vdb_res.VdbId
+
+	// poll for vdb
+	vdb_job_res, vdb_job_err := PollJobStatus(*vdb_res.Job.Id, context.Background(), client)
+
+	if vdb_job_res == Failed || vdb_job_res == Canceled || vdb_job_res == Abandoned {
+		println("An error occured during vdb job polling " + vdb_job_err)
+		return ""
+	}
+
+	// for eventual consistency
+	time.Sleep(20 * time.Second)
```

Comment:
Is this really required?

Reply:
Yes this is when DCT job is successful but resource takes few seconds to create in engine/DCT databases.. but I have changed this static wait function to existing PollForObjectExistence() to better match the other code.